components/renku_data_services/session/k8s_client.py

```diff
@@ -395,4 +279,35 @@ async def get_image_build_logs(self, buildrun_name: str, max_log_lines: int | No
         pod_name = taskrun.status.podName if taskrun.status else None
         if not pod_name:
             raise errors.MissingResourceError(message=f"The buildrun {buildrun_name} has no pod to retrieve logs from.")
-        return await self.base_client.get_pod_logs(name=pod_name, max_log_lines=max_log_lines)
+        return await self.get_pod_logs(name=pod_name, max_log_lines=max_log_lines)
+
+    async def get_pod_logs(self, name: str, max_log_lines: int | None = None) -> dict[str, str]:
```

Comment:
suggestion: I think we should make this private. It is not used anywhere else. And I dont see that anyone would call this directly. In this case we care about the logs of the image build.